tests/e2e/performance/mcg/test_list_objects.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.bucket_utils import copy_objects, craft_s3_command
+from ocs_ci.ocs.perftests import PASTest
+from ocs_ci.framework.testlib import performance, bugzilla, skipif_ocs_version
+from ocs_ci.ocs.resources.mcg import MCG
+
+
+log = logging.getLogger(__name__)
+
+LINUX_TAR = "https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.4.tar.xz"
+
+
+@performance
+@bugzilla("2052079")
+@skipif_ocs_version("<4.8")
+@pytest.mark.polarion_id("OCS-3926")
+class TestListOfObjects(PASTest):
+    """
+    Test to verify the list whole objects without any failures
+
+    """
+
+    def test_whole_list_of_objects(
+        self,
+        awscli_pod_session,
+        bucket_factory,
+    ):
+        """
+        Testcase to verify list whole objects lists without any failures
+
+        1. Create a OBC
+        2. Copied/Uploaded directory to the objects which
+           contains large number of directories, for e,g linux tar
+        3. List objects
+
+        """
+
+        # Create OBC
+        bucketname = bucket_factory(1)[0].name
+        full_object_path = f"s3://{bucketname}"
+        s3_obj = MCG(bucketname)
+
+        # Get linux tar and copy linux directory to the objects
+        data_dir = "linux_tar_dir/"
+        command = (
+            f"mkdir {data_dir}; cd {data_dir}; yum install wget -y; "
+            f"yum install tar -y; yum install xz -y; "
+            f"wget {LINUX_TAR}; tar -xvf linux-4.4.tar.xz; (for i in $(seq 1 2); "
+            f"do cp -r linux-4.4/ linux-$i; done)"
+        )
+        awscli_pod_session.exec_sh_cmd_on_pod(command=command, sh="sh")
+        files_count = awscli_pod_session.exec_sh_cmd_on_pod(
+            command=f"find  {data_dir} -type f -follow -print | wc -l", sh="sh"
+        )
+
+        copy_objects(
+            awscli_pod_session,
+            data_dir,
+            full_object_path,
+            s3_obj,
+            recursive=True,
+            timeout=1800,
+        )
+
+        # List the objects recursive
+        cmd = (
+            f"ls {full_object_path} --recursive | grep "
+            "linux"
+            "| sort| uniq -c| wc -l"
+        )
+        objects_count = awscli_pod_session.exec_cmd_on_pod(
+            command=craft_s3_command(cmd=cmd, mcg_obj=s3_obj), out_yaml_format=False
+        )
+
+        # Verify the whole sub directories are listed correctly
```

Comment:
Also, please find out the expected time to complete this listing and try to match it